src/components/PopoverMenu/index.js

```diff
@@ -1,136 +1,102 @@
-            this.attachKeyboardListener();
-        } else {
-            this.removeKeyboardListener();
-        }
-    }
+    /** The horizontal and vertical anchors points for the popover */
+    anchorPosition: PropTypes.shape({
+        horizontal: PropTypes.number.isRequired,
+        vertical: PropTypes.number.isRequired,
+    }).isRequired,
 
-    componentWillUnmount() {
-        this.removeKeyboardListener();
-    }
+    /** Where the popover should be positioned relative to the anchor points. */
+    anchorOrigin: PropTypes.shape({
+        horizontal: PropTypes.oneOf(_.values(CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL)),
+        vertical: PropTypes.oneOf(_.values(CONST.MODAL.ANCHOR_ORIGIN_VERTICAL)),
+    }),
+};
 
-    /**
-     * Set item to local variable to fire `onSelected` action after the menu popup closes
-     * @param {Object} item
-     */
-    selectItem(item) {
-        this.selectedItem = item;
-        this.props.onItemSelected(item);
-    }
+const defaultProps = {
+    ...createMenuDefaultProps,
+    anchorOrigin: {
+        horizontal: CONST.MODAL.ANCHOR_ORIGIN_HORIZONTAL.LEFT,
+        vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
+    },
+};
 
-    attachKeyboardListener() {
-        const shortcutConfig = CONST.KEYBOARD_SHORTCUTS.ENTER;
-        this.unsubscribeEnterKey = KeyboardShortcut.subscribe(
-            shortcutConfig.shortcutKey,
-            () => {
-                if (this.state.focusedIndex === -1) {
-                    return;
-                }
-                this.selectItem(this.props.menuItems[this.state.focusedIndex]);
-                this.setState({focusedIndex: -1}); // Reset the focusedIndex on selecting any menu
-            },
-            shortcutConfig.descriptionKey,
-            shortcutConfig.modifiers,
-            true,
-        );
-    }
+const PopoverMenu = (props) => {
+    const {isSmallScreenWidth} = useWindowDimensions();
 
-    removeKeyboardListener() {
-        if (!this.unsubscribeEnterKey) {
-            return;
-        }
-        this.unsubscribeEnterKey();
-    }
+    const selectItem = (index) => {
+        const selectedItem = props.menuItems[index];
+        props.onItemSelected(selectedItem);
+    };
 
-    resetFocusAndHideModal() {
-        this.setState({focusedIndex: -1}); // Reset the focusedIndex on modal hide
-        this.removeKeyboardListener();
-        if (this.selectedItem) {
-            this.selectedItem.onSelected();
-            this.selectedItem = null;
-        }
-    }
+    const [focusedIndex, setFocusedIndex] = useArrowKeyFocusManager({initialFocusedIndex: -1, maxIndex: props.menuItems.length - 1});
+    useKeyboardShortcut(
+        CONST.KEYBOARD_SHORTCUTS.ENTER,
+        () => {
+            if (focusedIndex === -1) {
+                return;
+            }
+            selectItem(focusedIndex);
+            setFocusedIndex(-1); // Reset the focusedIndex on selecting any menu
+        },
+        {isActive: props.isVisible},
+    );
```

Comment:
I think the changes in the code that cause this event listener to constantly resubscribe, make it climb to the top in the list of event handlers with the same priority. Since this event doesn't allow bubbling, it prevents any other handler. This causes this bug:

1. Open the panel to invite new members to a workspace (`/workspace/<workspace id>/invite`)
2. Verify that arrows up and down work and they change the selected row
3. Select something with the mouse
4. Click `Next`
5. Go back
6. BUG: The arrow keys don't work anymore


Listeners before re-subscribe:
![image](https://github.com/Expensify/App/assets/87341702/cdde6974-fb21-4db7-a2a9-8140ab4dcd7f)

Listeners after re-subscribe:

![image](https://github.com/Expensify/App/assets/87341702/966d4b9a-9503-4a82-80c3-a24490dda80c)